src/test/stub/rmc_vec.rs

```diff
@@ -0,0 +1,820 @@
+        unsafe {
+            self.len -= 1;
+            let result = read(self.ptr().offset(index as isize));
+            copy(
+                self.ptr().offset(index as isize + 1),
+                self.ptr().offset(index as isize),
+                self.len - index,
+            );
+            result
+        }
+    }
+
+    pub fn extend<I: Iterator>(&mut self, iter: I) where I: Iterator<Item = T> {
+        iter.for_each(move |element| {
+            self.push(element);
+        });
+    }
+
+    pub fn len(&self) -> usize {
+        self.len
+    }
+
+    pub fn reserve(&mut self, additional: usize) {
+        self.buf.reserve(self.len, additional);
+    }
+
+    pub fn reserve_exact(&mut self, additional: usize) {
+        self.buf.reserve(self.len, additional);
+    }
+
+    pub fn set_len(&mut self, new_len: usize) {
+        assert!(new_len <= self.buf.capacity());
+
+        self.len = new_len;
+    }
+
+    pub fn as_mut_ptr(&mut self) -> *mut T {
+        self.ptr()
+    }
+
+    pub fn as_slice(&self) -> &[T] {
+        self
+    }
+
+    pub fn as_mut_slice(&mut self) -> &mut [T] {
+        self
+    }
+
+    pub fn as_ptr(&self) -> *const T {
+        self.buf.ptr.as_const_ptr()
+    }
+
+    pub fn truncate(&mut self, len: usize) {
+        // TODO: For soundness wrt standard library, call drop_in_place() for pointer
+        // pointing to a slice of elements which were dropped.
+        self.len = len;
+    }
+
+    // Clears the vector, removing all values.
+    // This method has no effect on the allocated capacity of the vector
+    pub fn clear(&mut self) {
+        self.truncate(0);
+    }
+
+    pub fn dedup_by<F>(&mut self, same_bucket: F)
+    where
+        F: FnMut(&mut T, &mut T) -> bool {
+        assert!(false);
+    }
+
+    pub fn dedup_by_key<F, K>(&mut self, key: F)
+    where
+        F: FnMut(&mut T) -> K,
+        K: PartialEq<K> {
+        assert!(false);
+    }
+
+    // TODO:
+    // pub fn drain<R>(&mut self, range: R) -> Drain<'_, T, A>
+    // where
+    //     R: RangeBounds<usize> {
+    //     assert!(false);
+    // }
+
+    // TODO:
+    // pub fn drain_filter<F>(&mut self, filter: F) -> DrainFilter<'_, T, F, A>
+    // where:
+    //     F: FnMut(&mut T) -> bool {
+    //     assert!(false);
+    // }
+
+    // TODO:
+    // pub fn try_reserve(&mut self, additional: usize) -> Result<(), TryReserveError> {
+    //     unimplemeted!()
+    // }
+
+    // TODO:
+    // pub fn try_reserve_exact(&mut self, additional: usize) -> Result<(), TryReserveError> {
+    //     unimplemented!()
+    // }
+
+    pub fn swap_remove(&mut self, index: usize) -> T {
+        let len = self.len;
+        assert!(index < len);
+
+        unsafe {
+            let last = read(self.as_ptr().add(len - 1));
+            let hole = self.as_mut_ptr().add(index);
+            self.set_len(len - 1);
+            replace(hole, last)
+        }
+    }
+    pub fn capacity(&self) -> usize {
+        self.buf.capacity()
```

Comment:
Technically, this is not an over-approximation of what the standard library can do